core/node/node_sync/src/external_io.rs

```diff
@@ -337,35 +337,57 @@ impl StateKeeperIO for ExternalIO {
             .await
             .context("failed to fetch protocol version from the main node")?
             .context("protocol version is missing on the main node")?;
+        #[allow(deprecated)]
+        let (minor, bootloader_code_hash, default_account_code_hash) = {
+            let minor = protocol_version
+                .minor_version
+                .or(protocol_version.version_id)
+                .context("Missing minor protocol version")?;
+            let bootloader_code_hash = protocol_version
+                .bootloader_code_hash
+                .or_else(|| {
+                    protocol_version
+                        .base_system_contracts
+                        .map(|hashes| hashes.bootloader)
+                })
+                .context("Missing bootloader code hash")?;
+            let default_account_code_hash = protocol_version
+                .default_account_code_hash
+                .or_else(|| {
+                    protocol_version
+                        .base_system_contracts
+                        .map(|hashes| hashes.default_aa)
+                })
+                .context("Missing default account code hash")?;
```

Comment:
To avoid `allow(deprecated)` maybe it makes sense to add getters on `ProtocolVersion` that would encapsulate this logic?